idris-highlight-input.el

```diff
@@ -60,14 +60,14 @@ See Info node `(elisp)Overlay Properties' to understand how ARGS are used."
       (widen)
       (if (or (> end-line start-line)
               (and (= end-line start-line)
-                   (> end-col start-col)))
+                   (>= end-col start-col)))
           (with-current-buffer buffer
             (save-excursion
               (goto-char (point-min))
               (let* ((start-pos (+ (line-beginning-position start-line)
                                    (idris-highlight-column start-col)))
                      (end-pos (+ (line-beginning-position end-line)
-                                 (idris-highlight-column end-col)))
+                                 (idris-highlight-column (+ end-col 1))))
```

Comment:
I don't use Lisp very often. From a style perspective, since this is already part of a + expression, would this have been preferable?

```emacs
(end-pos (+ (line-beginning-position end-line)
            (idris-highlight-column end-col)
            1))
```

Reply:
seems to me shorter and equally readable so I would say yes, although I am no lisp expert myself.

Reply:
I actually think that the right thing to do is to have it say `(idris-highlight-column (+ end-col 1))`, or potentially `(idris-highlight-column (+1 end-col))`.

This is because `idris-highlight-column`, semantically speaking, is translating between the Idris's notion of column numbering and Emacs's, where one of them is zero-indexed and the other one-indexed. And the transformation here is applied to the incremented data from Idris, rather than transforming the whole thing.

But that's a very minor argument, and any of these solutions are really fine.

Reply:
We can merge as-is if you don't find me convincing here :-)

Reply:
@david-christiansen I'd prefer to be semantically correct over syntactically terse, so I've changed it once more.